okapi-core/src/main/java/org/folio/okapi/managers/ProxyService.java

```diff
@@ -174,6 +187,27 @@ private boolean resolveRedirects(ProxyContext pc,
     return true;
   }
 
+  private boolean checkTokenCache(ProxyContext pc, HttpServerRequest req, RoutingEntry re,
+      ModuleInstance mi) {
+    boolean skipAuth = false;
+    String pathPattern = re.getPathPattern();
+
+    CacheEntry cached = tokenCache.get(pc.getTenant(), req.method().name(),
+        pathPattern == null ? req.path() : pathPattern, req.getHeader(XOkapiHeaders.USER_ID),
+        req.headers().get(XOkapiHeaders.TOKEN));
+
+    if (cached != null) {
+      mi.setAuthToken(cached.token);
+      mi.setxOkapiUserId(cached.xokapiUserid);
+      mi.setxOkapiPermissions(cached.xokapiPermissions);
+
+      skipAuth = true;
+    } else {
+      mi.setAuthToken(req.headers().get(XOkapiHeaders.TOKEN));
```

Comment:
Can we be sure that there is no old userId and permission in mi? Or should we set them to null?

Reply:
done